autoload/ale/floating_preview.vim

```diff
@@ -0,0 +1,91 @@
+" Author: Jan-Grimo Sobez <jan-grimo.sobez@phys.chem.ethz.ch>
+" Author: Kevin Clark <kevin.clark@gmail.com>
+" Description: Floating preview window for showing whatever information in.
+
+" Precondition: exists('*nvim_open_win')
+
+function! ale#floating_preview#Show(lines, ...) abort
+    if !exists('*nvim_open_win')
+        execute 'echom ''Floating windows not supported in this vim instance.'''
+
+        return
+    endif
+
+    " Remove the close autocmd so it doesn't happen mid update
+    augroup NvimFloating
```

Comment:
We should prefix groups we create with `ALE` so we don't conflict with other groups.